### PR TITLE
Mike welcome

### PR DESCRIPTION
**Pull request by MikeSewell**
Updated readme and added the welcome new users feature

### CODE REVIEW

**Comment by krogers78** [comment]
Everything looks good. My one suggestion would be to maybe utilize the variable inside your function for the welcome message to personalize the welcome message, but if not then everything looks good!

**Comment by LevanceWam** [comment]
Hey mike code looks good I would add some more comments and for your controller name for your welcome file I would name that a little differently its bland and can easily confuse others. besides that Good code.

**Comment by MikeSewell** [comment]
@krogers78 That's a good idea, I didn't think about adding the name to the message that would be a good way to use the `this` in the method.

File: bot/controllers/newUsers.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const BaseController = require('../baseController.js');
+const Command = require('../baseCommand.js');
+
+class newUserController extends BaseController {
+  constructor(message) {
+    super(message);
+    const controller = this;
+    this.commands = [
+      new Command('!welcome', '!welcome', 'Welcome Message', 'Sends a new user of the channel the welcome message', this.msgNewUserWelcome.bind(controller), 'dm'),
```
Comment (reactivepixel) [inline bot/controllers/newUsers.js:9]: Your implementation is deviating from in the formatting from the other established controllers.

File: bot/client.js [hunk 2]
```diff
@@ -57,4 +57,10 @@ client.on('message', (message) => {
   }
 });
 
+client.on('guildMemberAdd', (member) => {
+  member.sendMessage('Welcome to the channel!');
```
Comment (reactivepixel) [inline bot/client.js:61]: Consider that your implementation solves the problem, and elegantly, but falls just short in that your solution ​within the client.js file does not implement the controller you created. So in effect, any further maintenance will require duplicate efforts in keeping both messages identical.

File: readme.md [hunk 3]
```diff
@@ -111,11 +101,11 @@ Removing a semi-colon from any line of Javascript will trigger the linter to aut
 ## Example of Usage
 
 Instead of console.log() require the tool.js file
-and call the .debug() method passing in three arguments
+and call the .log() method passing in three arguments
 (title, output, level)
 
-[code] to call a debug()
-[code] to activate debug_mode
+[code] to call a log()
```
Comment (reactivepixel) [inline readme.md:107]: Good catch.

**Review by reactivepixel — CHANGES_REQUESTED** [review]
Overall very good implementation, but you fall short of this code being ready for release due to the controller not being called / implemented in client.js

**Pull request closed without merging**